docs/bmo-part-of-capm3.md

```diff
@@ -0,0 +1,240 @@
+    ├── keepalived_patch.yaml
+    └── kustomization.yaml
+
+```
+
+Whereas, ironic-deployment folder has three top level folder for deployments,
+namely `default`,  `ironic` and `keepalived`. `default` and `ironic` deploy
+only ironic, `keepalived` deploys the ironic with keepalived. As the name
+implies, `keepalived/keepalived_patch.yaml` patches the default image URL
+through kustomization. The user should run following commands to be able to
+meet requirements of each use case as provided below:
+
+### Command to deploy "production" use case after structural changes
+
+```diff
+kustomize build $BMOPATH/deploy/default | kubectl apply -f-
+kustomize build $BMOPATH/ironic-deployment/default | kubectl apply -f-
+
+```
+
+### Command to deploy "local developer" use case after structural changes
+
+```diff
+kustomize build $BMOPATH/ironic-deployment/default | kubectl apply -f-
+
+```
+
+### Command to deploy "ironic outside of the cluster" use case after structural changes
+
+```diff
+kustomize build $BMOPATH/deploy/default | kubectl apply -f-
+
+```
+
+where $BMOPATH points to the baremetal-operator path.
+
+To this end, "production", "local developer" and "ironic outside of the
+cluster" use cases will be met, as new structure will fulfill the needs of those
+cases with cleaner and leaner organization of deployments in place.
+
+## Pivoting use case
+
+Once above outlined folder structural changes are made, we can deploy BMO on its
+own in CAPM3 and Ironic independently to meet the use case of pivoting, where
+BMO should be deployed as part of CAPM3. The main concern here is that, BMO and
+Ironic are separated and as a consequence, it will result in breaking the chain
+of workflow, where, previously, BMO could use localhost to talk to ironic
+without any authentication. However, considering that Ironic runs in host
+network, BMO would be able to interact with it, from anywhere that is connected
+or routed to the provisioning network. Therefore, separating Ironic in its own
+pod should not impact BMO in this case.
+
+Furthermore, to support above provided statement, CAPM3  needs to embed BMO in a
+way that it is deployed and modified properly by `clusterctl` without having to
+embed ironic as part of the clusterctl. Thus, we created
+`cluster-api-provider-metal3/config/bmo/` folder to be able to deploy BMO in
+CAPM3. The following directory tree and kustomization file will give more
+explanation on what are we trying to achieve as an end-goal.
+
+```diff
+tree config/bmo/
+
+config/bmo/
+├── bmo_image_patch.yaml
+├── kustomization.yaml
+└── README.md
+
+```
+
+where `kustomization.yaml` file contains:
+
+```yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+namespace: capm3-system
+resources:
+- github.com/metal3-io/baremetal-operator/deploy/operator/?ref=master
+- github.com/metal3-io/baremetal-operator/deploy/crds/?ref=master
+- github.com/metal3-io/baremetal-operator/deploy/rbac/?ref=master
```

Comment:
Myabe we could release a deployment file for BMO, the same way we do for ip-address-manager, and import that to deploy a "released" version of BMO with a specific image tag instead of always taking master in. Just a thought, that should probably be tackled as part of the release process.

Reply:
Are you referring to this [one](https://github.com/metal3-io/ip-address-manager/blob/382649b7f3f06bd2b06950eb5543c03cfc130dd0/docs/releasing.md#creating-a-release-for-ipam)? Should I add then a release document separately?

Reply:
no, there's no need right now to change this, but we could discuss this as part of the release process of BMO